### PR TITLE
Plaz check for existence of Bearbeitungsvermerke when enum sonstige anywhere

### PR DESCRIPTION
**Pull request by mariusheine**

### CODE REVIEW

**Comment by TruongQuangSB** [comment]
I think, u can filter _BasisAttribut_AttributeGroup_ by _allContents_, then find function _getWert_ with method _declaredMethods_ from Class and u can check, if the _Wert_ is _Enumerator_ or not. By that u don't need a List from BasisObject, which contains Enumerator

**Comment by TruongQuangSB** [comment]
In addition we prefer to implement in Java instead of Xtext because we want to gradually move away from Xtext

File: java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * Validates that any enum value "Sonstige" has an according bearbeitungsvermerk anywhere
+ * 
+ * @author mariusheine
+ */
+@Component
+class SonstigeRequiresBearbeitungsvermerk implements PlazCheck {	
+	final Check<Basis_Objekt>[] RELEVANT_ENTITIES = #[
+		new Check(Aussenelementansteuerung, [ aussenElementAnsteuerungArt ], "aussenelementansteuerungArt"),
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend:35]: Listing all possible types with enumerations here seems like a rather tedious task.

My suggestion would be to iterate all elements (eAllContents) and then filter for entries with a value that has an `eClass` that is an `EEnum`. That way we can identify every enumeration value in the planning file.

For those set to a Sonstige value, we can then go back up the EMF parent chain until we find the containing 
`Ur_Objekt` and check it for a remark.

**Review by pstuecker — CHANGES_REQUESTED** [review]

File: java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend (hunk 1 above)
Comment (mariusheine) [reply java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend:35]: @pstuecker This does not work. the `eClass` is never an instance of or equal to `EEnum`.

I try it now with the idea of @TruongQuangSB  https://github.com/eclipse-set/set/pull/676#issuecomment-2175843950

File: java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		new Check(Uebertragungsweg, [ bandbreite ], "aussenelementansteuerungArt"),
+		new Check(Bahnsteig_Zugang, [ bahnsteigZugangArt ], "bahnsteigZugangArt")
+	];
+	
+	override run(IModelSession modelSession) {
+		return RELEVANT_ENTITIES.flatMap([
+			return it.singleRun(modelSession.planProSchnittstelle.eAllContents)
```
[the remark this answers is not in the document]
Comment (mariusheine) [reply java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend]: Changed it java class so this doesn't matter anymore

File: java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend (hunk 1 above)
Comment (mariusheine) [reply java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SonstigeRequiresBearbeitungsvermerk.xtend]: This works now nicely with a combined way

**Comment by mariusheine** [comment]
@pstuecker I also tried to achieve this grouping of errors like for the "Teilbereichsgrenzen" but was not able to achieve this. Maybe you can give me a hint on how to do this

**Comment by pstuecker** [comment]
It actually already works, it just doesn't show up in PHausen, as groups are only created if all fields (excluding row number, line number and message) are equal, and there are at least 5 elements in the group. In PHausen all groups are of size 4 or less.

**Review by pstuecker — APPROVED** [review]

**Pull request merged**